### ISSUE TITLE
🐛 [BUG] - KubeMonitoring chart produces a diff on every helm upgrade

### ISSUE THREAD

**Issue by IvoGoman** [cloudoperators/greenhouse-extensions#547]
### Priority

(Medium) I'm annoyed but I'll live

### Description

Related to #546 

The `kube-monitoring` plugin creates a diff on every helm diff, this leads to reoccurring reconciliations. 
The problematic file is `Secret/tls-prometheus-alertmanager-auth` which looks to be static in the chart: https://github.com/cloudoperators/greenhouse-extensions/blob/main/kube-monitoring/charts/templates/alertmanager-tls-secret.yaml.

The secret values used are mounted via secretRef into the Plugin. In this case from `tls-greenhouse-prometheus-auth`, which is the secret for Certificate `greenhouse-prometheus-auth`. 
https://github.com/cloudoperators/greenhouse-extensions/blob/c8fbfb1c1bdb636cda6c9e838032a490181b6186/alerts/charts/templates/ca-secret-issuer-cert.yaml#L38-L39

Since the Certificate uses helm lifecycle hooks and no `"helm.sh/hook-delete-policy"` is defined, it is recreated on every helm deploy [If no hook deletion policy annotation is specified, the before-hook-creation behavior applies by default.](https://helm.sh/docs/topics/charts_hooks/#hook-deletion-policies)

```text
"Secret/tls-prometheus-alertmanager-auth: [\n {\n \"value\": \"***** - after\",\n \"op\": \"replace\",\n \"path\": \"/data/tls.crt\"\n },\n {\n \"value\": \"***** - after\",\n \"op\": \"replace\",\n \"path\": \"/data/tls.key\"\n }\n]"} 
```

### Reproduction steps

```bash
1. Go to the organisation namespace 
2. `helm history kube-monitoring`
3. see frequent upgrades of the plugin
```

### Screenshots

```bash
![DESCRIPTION](LINK.png)
```

**Comment by richardtief** [comment]
@IvoGoman I think this is solved by keeping the cert in `greenhouse-prometheus-auth` generated by the alerts plugin static.

**Comment by IvoGoman** [comment]
Ah yes, I overlooked that they are part of the same condition. Can be closed in favour of #546
